aten/src/ATen/LegacyBatchedTensorImpl.cpp

```diff
@@ -124,15 +124,15 @@ Tensor makeBatched(const Tensor& tensor, BatchDims bdims) {
       std::all_of(bdims.begin(), bdims.end(),
           [](const BatchDim& bdim) { return bdim.level() < kVmapNumLevels; }),
       "We only support up to ", kVmapNumLevels, " nested vmaps");
-  return at::detail::make_tensor<BatchedTensorImpl>(tensor, std::move(bdims));
+  return at::detail::make_tensor<BatchedTensorImpl>(std::move(tensor), std::move(bdims));
 }
 
-Tensor addBatchDim(const Tensor& tensor, int64_t level, int64_t dim) {
+Tensor addBatchDim(Tensor tensor, int64_t level, int64_t dim) {
   const auto* batched = maybeGetBatchedImpl(tensor);
   if (!batched) {
     BatchDims bdims;
     bdims.emplace_back(level, dim);
-    return at::detail::make_tensor<BatchedTensorImpl>(tensor, std::move(bdims));
+    return at::detail::make_tensor<BatchedTensorImpl>(std::move(tensor), std::move(bdims));
```

Comment:
make_tenosr actually efficiently uses an rvalue? Just want to make sure since this is a big change to the rest of the codebase here.

Reply:
BatchedTensorImpl has a tensor as member.